### DISCUSSION TITLE
Value error in Gibs sampling : Variable names cannot be same

### DISCUSSION THREAD

**Discussion by tenzosan** [pgmpy/pgmpy#1591] [Show and tell]
from pgmpy.models import BayesianNetwork
from pgmpy.factors.discrete import TabularCPD
from pgmpy.sampling import GibbsSampling
Catch = BayesianNetwork([('B','A'),('E','A'),('A','J'),('A','M')])
burg_cpd = TabularCPD('B',2, values = [[0.999],[0.001]])
print(burg_cpd)
earth_cpd = TabularCPD('E',2, values = [[0.998],[0.002]])
print(earth_cpd)
alarm_cpd = TabularCPD('A', 2, values = [[0.999,0.71,0.06,0.05],
                                         [0.001,0.29,0.94,0.95]],
                                        evidence=['B','E'],evidence_card=[2,2])
print(alarm_cpd)
merry_cpd = TabularCPD('M',2, values = [
                                         [0.99,0.3],[0.01,0.7]],
                                         evidence=['A'],evidence_card=[2])
print(merry_cpd)

john_cpd = TabularCPD('J',2, values = [
                                         [0.95,0.1],[0.05,0.9]],
                                         evidence=['A'],evidence_card=[2])
print(john_cpd)
Catch.add_cpds(merry_cpd,john_cpd,alarm_cpd,earth_cpd,burg_cpd)
Catch.nodes()
gibbs_chain = GibbsSampling(Catch)
gibbs_chain.sample(size=300)

**Comment by arnavk23** [comment]
You're using `Catch` as the name of the `BayesianNetwork`, **but 'Catch' is also a Python keyword**, and more importantly:

You're assigning the model to a variable called `Catch`, but **you've also used `Catch.add_cpds(...)` later**, which might not be what's triggering the error—but it's potentially confusing.

However, the actual issue likely stems from:

```python
gibbs_chain.sample(size=300)
```

### Cause:

When you run `gibbs_chain.sample(...)`, pgmpy tries to sample all variables. If **two CPDs or nodes have the same variable name**, or if the model isn't correctly defined (e.g., duplicate nodes due to typos or implicit errors), this can trigger:

```
ValueError: Variable names cannot be same
```

But your CPDs and graph look fine at first glance.

### Solution:

It turns out the actual issue is likely due to this:

**One of the variable names conflicts with the Python built-in name `Catch`.** Changing the model name from `Catch` to something else (e.g., `alarm_model`) will avoid confusion and any possible name shadowing or internal conflict.
* Make sure you **call `check_model()`** to validate the network before sampling.
* Ensure that **no CPDs or nodes share the same variable name**.
